magneticod/magneticod/__main__.py

```diff
@@ -55,24 +52,22 @@
 def main():
     global complete_info_hashes, database, node, peers, selector
 
-    logging.basicConfig(level=logging.INFO, format="%(asctime)s  %(levelname)-8s  %(message)s")
-    logging.info("magneticod v%d.%d.%d started", *__version__)
-
     arguments = parse_cmdline_arguments()
-    if arguments is None:
-        return 2
+
+    logging.basicConfig(level=arguments.loglevel, format="%(asctime)s  %(levelname)-8s  %(message)s")
+    logging.info("magneticod v%d.%d.%d started", *__version__)
 
     # noinspection PyBroadException
     try:
-        path = os.path.join(appdirs.user_data_dir("magneticod"), "database.sqlite3")
+        path = arguments.database_file
         database = persistence.Database(path)
     except:
         logging.exception("could NOT connect to the database!")
         return 1
 
     complete_info_hashes = database.get_complete_info_hashes()
 
-    node = dht.SybilNode(arguments.node_addr)
+    node = dht.SybilNode(arguments.node_addr, max_metadata_size=arguments.metadata_size_limit)
     node.when_peer_found = on_peer_found
```

Comment:
Use [`functools.partial`](https://docs.python.org/3.5/library/functools.html#functools.partial) here. I think it would look like this (did **not** test it):

    node.when_peer_found = functools.partial(on_peer_found(arguments.metadata_size_limit))

And you need to change the signature of the function to make the `max_metadata_size` the first argument.